setup.py

```diff
@@ -1,5 +1,18 @@
 from pathlib    import Path
 from setuptools import setup
+import sys
+
+min_version = (3, 6)
+
+if sys.version_info < min_version:
+    error = """
+Beginning with Matplotlib 3.1, Python {0} or above is required.
+
+This may be due to an out of date pip.
+
+Make sure you have pip >= 9.0.1.
```

Comment:
Is this message something specific to matplotlib or is it generally an issue users encounter with pip and changing between Python versions?

Reply:
This is a general problem with old versions of pip not knowing about the require_python meta data